src/config/scrivitoContentBrowser.js

```diff
@@ -1,107 +1,152 @@
 import * as Scrivito from "scrivito";
 
 Scrivito.configureContentBrowser({
-  filters: {
+  filters: filterContext => {
+    const validObjClasses = filterContext._validObjClasses;
+    if (validObjClasses) {
+      switch (validObjClasses.length) {
+        case 0:
+          return invalidObjClassFilter();
+        case 1:
+          return objClassFilter(validObjClasses[0]);
+        default:
+          return objClassesFilter(validObjClasses);
+      }
+    }
+
+    return defaultFilter();
+  },
+});
+
+function invalidObjClassFilter() {
+  return {
+    _objClass: {
+      options: {
+        All: {
+          title: "Invalid restrictions",
+          selected: true,
+        },
+      },
+    },
+  };
+}
+
+function objClassFilter(objClass) {
+  return {
+    _objClass: {
+      options: {
+        [objClass]: {
+          ...descriptionForObjClass(objClass),
+          selected: true,
+        },
+      },
+    },
+    _modification: MODIFICATION_FILTER,
+  };
+}
+
+function objClassesFilter(objClasses) {
+  return {
     _objClass: {
       options: {
         All: {
           title: "All",
           icon: "folder",
-          query: Scrivito.Obj.all(),
+          field: "_objClass",
+          value: objClasses,
           selected: true,
         },
-        Images: {
-          title: "Images",
-          icon: "image",
-          field: "_objClass",
-          value: "Image",
+        ...objClasses.reduce((interimOptions, item) => {
+          return {
+            ...interimOptions,
+            ...{ [item]: descriptionForObjClass(item) },
+          };
+        }, {}),
+      },
+    },
+    _modification: MODIFICATION_FILTER,
```

Comment:
* [ ] FYI we could prevent some repetition by making the `MODIFICATION_FILTER` include the filter key, and adding it by `...MODIFICATION_FILTER`. After that we could even go one step further and give it a more general name like `MORE_FILTERS` (which is how it appears in the UI).

Reply:
I thought this was too ES6, but several lines above the spread is already in use, therefore 👍